src/feature/relay/circuitbuild_relay.c

```diff
@@ -123,31 +123,38 @@ circuit_extend_add_ed25519_helper(struct extend_cell_t *ec)
  * and are allowed by the current ExtendAllowPrivateAddresses config.
  *
  * If they are valid, return 0.
- * Otherwise, if they are invalid, log a warning at <b>log_level</b>,
- * and return -1.
+ * Otherwise, if they are invalid, return -1.
+ * If <b>log_zero_addrs</b> is true, log warnings about zero addresses at
+ * <b>log_level</b>. If <b>log_internal_addrs</b> is true, log warnings about
+ * internal addresses at <b>log_level</b>.
  */
 static int
 circuit_extend_addr_port_helper(const struct tor_addr_port_t *ap,
```

Comment:
We should probably rename this function so that it reflects the predicate that it evaluates.

Reply:
Renamed and returned bool in 496ddd2877051a43aaa7bbeb68bd5b255c71adc7. The code is much nicer now.